### DISCUSSION TITLE
自动标题识别不准确，总是识别为二级

### DISCUSSION THREAD

**Discussion by JackTvT** [hugo-next/hugo-theme-next#72] [Q&A]
请按照此 Issue 模版提供相关信息，例如源码仓库、网站链接和屏幕截图，这将有助于我们进行调查。
(Please follow this Issue template to provide relevant information, such as source code repository, website URL and screenshots, which will help us investigate.)

## 问题自检列表 (Issue Checklist)

> 请确认如下情况已检查，并将 [ ] 换成 [x] 来选择 (Change [ ] to [x] to selected which your could make sure)

- [x] 已使用 Hugo Extended 0.86.0 或更高的版本 (I'm using Hugo Extended version 0.86.0 or later)
- [x] 已使用 Hugo NexT 4.0 或更高的版本 (I'm using Hugo NexT version 4.0 or later)
- [x] 已阅读 [Hugo 故障排除](https://gohugo.io/troubleshooting/)并尝试修复无果 (I had already read the [Troubleshooting page of Hugo](https://gohugo.io/troubleshooting/))
- [x] 已搜索过当前所有[已知问题](https://github.com/hugo-next/hugo-theme-next/issues)和[讨论区](https://github.com/hugo-next/hugo-theme-next/discussions/categories/q-a) ，但没有找到帮助 (I had already searched for current [Issues](https://github.com/hugo-next/hugo-theme-next/issues) and [Q & A](https://github.com/hugo-next/hugo-theme-next/discussions/categories/q-a), which does not help me.)

***

## 预期行为 (Expected behavior)

> 请描述你想实现的预期效果 (Please describe the expected behavior what you want)
可以配置一个自动标题目录开关，新建hugo博客的使用自动生成，从别的主题转过来就用之前的，这样不用每篇文章都改。
另外，如果可以添加Header部分放置菜单栏，侧边栏就放置个人信息和文章信息就更好了

## 实际行为 (Actual behavior)

> 请同时提供网站链接、屏幕截图和源码地址 (Please provide your site links, screenshots and code repository)
![image](https://user-images.githubusercontent.com/56101781/203241840-4f804437-1cff-47f8-bf35-d10c8b0857ca.png)

### 主题配置(Configuration)

> 提供主题 `config.yaml` 中修改过的部分内容 (Paste ONLY CHANGED CONFIGURATION from  `config.yaml`)

```yaml

```

## 其他有用信息 (Other Information)

> 例如：操作系统、浏览器等 (e.g. Browser, System)

**Comment by elkan1788** [comment]
自动标题识别是以H2开始的，现在暂不支持从H3开始，原因有2点：
1. Hugo 默认文章内的标题都是从H2开始，而且这部分的配置暂没办法获取
2. 文章的Title已经使用H1，再增加H1在有些网页评测机构认为设计不规范。

后续会考虑和Hugo的配置进行搭配，让用户可以自定义统计开始的标题字体大小。

**Reply by JackTvT** [reply]
好的，感谢解答，关于自定义配置这个，我是想表达从他主题或平台转过来，md中可能已经写了一、二、三这些序号，这样的话再自动生成就会多余，需要挨个删除原有的序号，所以希望有个关闭自动统计的功能

**Reply by elkan1788** [reply]
> 好的，感谢解答，关于自定义配置这个，我是想表达从他主题或平台转过来，md中可能已经写了一、二、三这些序号，这样的话再自动生成就会多余，需要挨个删除原有的序号，所以希望有个关闭自动统计的功能

可以支持关闭自动目录的功能，只需要在配置文件中，把如下参数设置为 false 即可：

```
# 文章目录显示
  # Table of Contents in the Sidebar
  # Front-matter variable (nonsupport wrap expand_all).
  toc:
    enable: true
    # 自动设置列表的数字
    # Automatically add list number to toc.
    number: false
```
